### PR TITLE
Add usePrivateIP for cloudsql postgresql instances 

### PR DESCRIPTION
**Pull request by KyriosGN0**
🛠️ Description

add the ability to speificy if the cloudsql postgresql instance has private ip we should use 
right now we fail if the instance has only public ip 

Closes [#26827](https://github.com/hashicorp/vault/issues/26827)

### CODE REVIEW

**Comment by sviatlo** [comment]
Hello @hsimon-hashicorp!
We've been dreaming of using Vault to authenticate with gcp sql. Is there any chance you can help to facilitate/expedite the review of this PR? 
Thank you in advance!

**Comment by hsimon-hashicorp** [comment]
> Hello @hsimon-hashicorp! We've been dreaming of using Vault to authenticate with gcp sql. Is there any chance you can help to facilitate/expedite the review of this PR? Thank you in advance!

Hi there! Thank you for the ping (genuinely, I appreciate it!) as well as your enthusiasm for this PR. I will check with our engineering and product management teams to see if we can get some eyes on this. I can't make any guarantees about timelines, but this does give me more data about community sentiment which is very valuable to me. Thanks again! ^_^

File: sdk/database/helper/connutil/sql.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ type SQLConnectionProducer struct {
 	AuthType                 string      `json:"auth_type" mapstructure:"auth_type" structs:"auth_type"`
 	ServiceAccountJSON       string      `json:"service_account_json" mapstructure:"service_account_json" structs:"service_account_json"`
 	DisableEscaping          bool        `json:"disable_escaping" mapstructure:"disable_escaping" structs:"disable_escaping"`
+	usePrivateIp             bool        `json:"use_private_ip" mapstructure:"use_private_ip" structs:"use_private_ip"`
```
Comment (robmonte) [inline sdk/database/helper/connutil/sql.go:43]: Nit: Rename this to be `usePrivateIP` to keep the acronym in capital letters to match others?

**Review by robmonte — COMMENTED** [review]
Hi @KyriosGN0
Thanks for the contribution!

In addition to the nit I left in the code, two additions are needed.

1. A changelog file needs to be added. \
  A new file named `26828.txt` should be placed in the `changelog` directory with the following contents:

  ````
  ```release-note:improvement
  secrets/database: Add support for GCP CloudSQL private IP's.
  ```
  ````

2. Documentation updates are needed. \
  The "Configure connection" section of the API docs located [here](https://developer.hashicorp.com/vault/api-docs/secret/databases/postgresql#configure-connection) will need to be updated to include this new configuration field. That file is located at `website/content/api-docs/secret/databases/postgresql.mdx` \
\
  The Setup portion of the "Authenticating to Cloud DBs via IAM" section of the feature documentation located [here](https://developer.hashicorp.com/vault/docs/secrets/databases/postgresql#setup-1) will need to be updated to include its use. This can be as simple as adding a new line to both CLI command examples to say `use_private_ip="false"`

File: sdk/database/helper/connutil/sql.go (hunk 1 above)
Comment (KyriosGN0) [reply sdk/database/helper/connutil/sql.go]: fixed !

**Comment by KyriosGN0** [comment]
hey @robmonte, Thanks you for the review, i addressed the nit in the code and added the required docs changes

File: sdk/database/helper/connutil/sql.go [hunk 2]
```diff
@@ -140,7 +141,7 @@ func (c *SQLConnectionProducer) Init(ctx context.Context, conf map[string]interf
 		// however, the driver might store a credentials file, in which case the state stored by the driver is in
 		// fact critical to the proper function of the connection. So it needs to be registered here inside the
 		// ConnectionProducer init.
-		dialerCleanup, err := c.registerDrivers(c.cloudDriverName, c.ServiceAccountJSON)
+		dialerCleanup, err := c.registerDrivers(c.cloudDriverName, c.ServiceAccountJSON, c.usePrivateIp)
```
Comment (robmonte) [inline sdk/database/helper/connutil/sql.go:144]:
```suggestion
		dialerCleanup, err := c.registerDrivers(c.cloudDriverName, c.ServiceAccountJSON, c.usePrivateIP)
```

Just this last tweak needed due to the field name change!

**Review by robmonte — APPROVED** [review]

**Comment by KyriosGN0** [comment]
@robmonte could you trigger the tests again ?

**Pull request merged**